docs/modules/ROOT/pages/how-tos/worklows/git-submodules.adoc

```diff
@@ -0,0 +1,128 @@
+= Git Submodule guideline
+
+This document outlines some basic guidelines for synchronizing changes between an upstream and downstream repository using Git submodules.
+
+By leveraging Git submodules, it is possible to maintain a consistent and up-to-date relationship between related repositories (upstream and downstream) while having more control of how the project is built, tested and released.
+
+Additionally, the integration with Konflux can automate key aspects of this process, ensuring that changes from the upstream repository are efficiently and reliably propagated to the downstream repository.
+
+== Implementing this guideline
+=== Basic definitions
+
+- **Upstream repo**: a community project which you do not have push access to (or which you do not want to push CI changes into).
+
+- **Downstream repo**: a self-owned copy of the upstream repository whose build, test, release lifecycle you want to control completely.
+
+=== Create a downstream repo
+
+The downstream repo can be used to do the following:
+
+* hold the git submodule reference to the upstream repository.
+* use the contents of the git submodule in a Containerfile for further edits.
+* onboarded onto {ProductName}.
+
+==== Configuring components
+
+Here is an example of a component definition:
+
```

Comment:
I think that this is more about defining components to use the proper contexts than it is just about the component definition. Since that is the case, I think that we can also change the content here:

* Describe the process to ensure the context is defined by the UI
* Describe the process to ensure that the context is defined by the CR definition

That being said, I think that this information would be better placed _in_ the documentation for defining the Component and this documentation would just be something like 

```
+
CAUTION: Make sure to properly set the xref:/path/to/doc.adoc#section[context and containerfile path] as required for the artifact build.
```

The context _may_ be just the submodule or it _may_ be `.` depending on whether additional content is defined in the repo (i.e. to help with migration of data).

Reply:
Good idea, I removed this section for now and referenced the docs on creating applications and components. Those details should be put there instead.

Reply:
Would you mind adding that information to the creating pages and then linking to it?

Reply:
I do not mind, I'll add it now and link back to it here

Reply:
Doc section linked: https://github.com/konflux-ci/docs/pull/120/files#diff-46f8b5b0ae481a31c84644c1fde822df6be41954ac5d4002de9444fe1b91fd08R26